couchbase/transactions/transactions_config.hxx

```diff
@@ -175,7 +226,34 @@ class transactions_config
         return *cleanup_hooks_;
     }
 
-  protected:
+    /** @internal */
+    struct built {
+        built();
+        ~built();
+        built(const built&);
+        built(const couchbase::durability_level,
+              const std::chrono::nanoseconds,
+              const std::optional<std::chrono::milliseconds>,
+              const std::unique_ptr<couchbase::core::transactions::attempt_context_testing_hooks>&,
+              const std::unique_ptr<couchbase::core::transactions::cleanup_testing_hooks>&,
+              const std::optional<transaction_keyspace>&,
+              const transactions_query_config::built&,
+              const transactions_cleanup_config::built&);
+        built& operator=(const built&);
```

Comment:
the constructors and destructors might be removed it. Let compiler to generate defaults.

The idea is that this "struct built" should be simple, without any methods, just plain data. And it is the "build()" method below, that initializes everything.

Reply:
I had compile issues with the unique_ptr to the testing hooks (which are forward declared), and added this stuff...  But perhaps I need to look at that again.

Reply:
After some thought, I realized a shared_ptr is just fine.  That eliminates the extra stuff in built.